msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/usgov/arlington/TestCase940421Clone.java

```diff
@@ -0,0 +1,130 @@
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadPromptHandler;
+import com.microsoft.identity.internal.testutils.labutils.LabConfig;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+
+// Interactive token acquisition with instance_aware=true and with custom claims request requiring
+// device auth {"access_token":{"deviceid":{"essential":true}}}
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/940421
+public class TestCase940421Clone extends AbstractMsalBrokerTest {
+
+    @Test
+    public void test_940421() throws Throwable {
+        final String username = mLoginHint;
+        final String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+
+        final MsalSdk msalSdk = new MsalSdk();
+
+        // create claims request object
+        final ClaimsRequest claimsRequest = new ClaimsRequest();
+        final RequestedClaimAdditionalInformation requestedClaimAdditionalInformation =
+                new RequestedClaimAdditionalInformation();
+
+        requestedClaimAdditionalInformation.setEssential(true);
+
+        // request the deviceid claim in ID Token
+        claimsRequest.requestClaimInIdToken("deviceid", requestedClaimAdditionalInformation);
+
+        final MsalAuthTestParams authTestParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(username)
+                .claims(claimsRequest)
+                .scopes(Arrays.asList(mScopes))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        // start interactive acquire token request in MSAL (should succeed)
+        final MsalAuthResult authResult = msalSdk.acquireTokenInteractive(authTestParams, new OnInteractionRequired() {
+            @Override
+            public void handleUserInteraction() {
+                final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                        .prompt(PromptParameter.SELECT_ACCOUNT)
+                        .loginHint(username)
+                        .sessionExpected(false)
+                        .consentPageExpected(false)
+                        .speedBumpExpected(false)
+                        .broker(mBroker)
+                        .expectingBrokerAccountChooserActivity(false)
+                        .expectingLoginPageAccountPicker(false)
+                        .registerPageExpected(true)
+                        .build();
+
+                new AadPromptHandler(promptHandlerParameters)
+                        .handlePrompt(username, password);
+            }
+        }, TokenRequestTimeout.MEDIUM);
+
+        authResult.assertSuccess();
```

Comment:
Just asserting success won't be enough here...see original test case for reference....in this test case we also need to assert that we got the `deviceid` claim back in the ID Token claims....previously this assertion was happening in the callback...now it needs to happen here (probably right after this line you can just add logic to assert for the claim)

Reply:
add the deviceid assert function